src/block.json

```diff
@@ -13,39 +13,41 @@
 	"attributes": {
 		"countryCode": {
 			"type": "string",
-			"source": "html",
-			"selector": ".xwp-country-card__country-code"
+			"source": "attribute",
+			"selector": ".xwp-country-card__country-code",
+			"attribute": "data-country-code"
 		},
 		"relatedPosts": {
 			"type": "array",
 			"source": "query",
-			"selector": ".related-post",
+			"selector": ".xwp-country-card__related-post",
 			"query": {
 				"title": {
 					"type": "string",
-					"selector": ".title",
+					"selector": ".xwp-country-card__related-post-title",
 					"source": "text"
 				},
 				"excerpt": {
 					"type": "string",
-					"selector": ".excerpt",
+					"selector": ".xwp-country-card__related-post-excerpt",
 					"source": "text"
 				},
 				"link": {
 					"type": "string",
-					"selector": ".link",
+					"selector": ".xwp-country-card__related-post-link",
 					"source": "attribute",
 					"attribute": "href"
 				},
 				"id": {
 					"type": "string",
-					"selector": ".link",
+					"selector": ".xwp-country-card__related-post-link",
 					"source": "attribute",
 					"attribute": "data-post-id"
 				}
 			}
 		}
 	},
+	"usesContext": [ "postId" ],
```

Comment:
@cdevroe Can you tell more about block context? How does block context work internally? Is the context available only in React or in PHP, too?

Reply:
As I understand it, Block context works similarly to how Reacts global context works. If you need nested components to retrieve some State, rather than needing to propagate all the way through the tree you can use context. I would likely use this feature sparingly. But I thought using context to determine the currently loaded postId was a good use of this feature, rather than relying on the URL to have the postId.

I didn’t prove this out in code, but I believe if we were using a render callback function in PHP to display the block on the front end we could access the postId using the block context.